### PR TITLE
Baseframe Emoji Validation

### PR DESCRIPTION
**Pull request by qurbat**
This PR adds the following:

 - ```emoji``` requirement to ```setup.py```
 - validation functionality to ```validators.py``` and ```fixtures.py```
 - test for validation to ```test_validators.py```

### CODE REVIEW

File: baseframe/forms/validators.py [hunk 1]
```diff
@@ -299,7 +300,23 @@ class NotEqualTo(_Comparison):
     def compare(self, value, other):
         return value != other
 
+class IsEmoji(object):
+    """
+    Validate that field.data does not belong to a public email domain.
+    If the domain lookup fails and mxsniff raises ``MXLookupException``, this validator
+    will still pass, as we expect that most domains are not public email domains.
```
Comment (jace) [inline baseframe/forms/validators.py]: Wrong docstring.

File: baseframe/forms/validators.py [hunk 2]
```diff
@@ -299,7 +300,23 @@ class NotEqualTo(_Comparison):
...
+
+    :param message:
+        Error message to raise in case of a validation error.
+    """
+    def __init__(self, message=None):
+        self.message = message or _(u'This is not a valid emoji.')
```
Comment (jace) [inline baseframe/forms/validators.py]: Message should be in double quotes. Our convention is to use double quotes for UI strings that need translation, and single quotes for tokens where the exact string must be preserved.

File: baseframe/forms/validators.py [hunk 3]
```diff
@@ -299,7 +300,23 @@ class NotEqualTo(_Comparison):
...
 
+    def __call__(self, form, field):
+        if field.data != "":
```
Comment (jace) [inline baseframe/forms/validators.py]: Just `if field.data:` should do.

File: baseframe/forms/validators.py [hunk 4]
```diff
@@ -299,7 +300,23 @@ class NotEqualTo(_Comparison):
...
+            if not field.data in emoji.UNICODE_EMOJI:
```
Comment (jace) [inline baseframe/forms/validators.py]: You can collapse these if statements into one: `if field.data and field.data not in emoji.UNICODE_EMOJI:`

File: tests/fixtures.py [hunk 5]
```diff
@@ -57,6 +57,13 @@ class UrlFormTest(forms.Form):
         filters=[forms.filters.strip()],
     )
 
+class EmojiFormTest(forms.Form):
+    emoji = forms.StringField(
+        __("Emoji"),
+        validators=[
+        forms.validators.IsEmoji()
```
Comment (iambibhas) [inline tests/fixtures.py]: this doesn't seem black compliant. If you have black setup, format this file once.

File: tests/test_validators.py [hunk 6]
```diff
@@ -42,6 +44,18 @@ def test_invalid_url(self):
             url = 'https://hasgeek'
             self.form.process(url=url)
             assert not self.form.validate()
+            
+    def test_valid_emoji(self):
+        with self.app.test_request_context('/'):
+            dat = u'\u2604'
```
Comment (iambibhas) [inline tests/test_validators.py]: You can also try giving an actual emoji as input to see if it works, like `😱`.

Comment (qurbat) [reply tests/test_validators.py]: Added! Seems to work.

File: baseframe/forms/validators.py (hunk 2 above)
Comment (iambibhas) [reply baseframe/forms/validators.py]: @qurbat this is not fixed.

Comment (qurbat) [reply baseframe/forms/validators.py]: @iambibhas fixed now

**Pull request merged**